src/core_plugins/kibana/server/lib/export/collect_index_patterns.js

```diff
@@ -0,0 +1,27 @@
+export default function collectIndexPatterns(savedObjectsClient, panels) {
+  const docs = panels.reduce((acc, panel) => {
+    const { kibanaSavedObjectMeta, savedSearchId } = panel.attributes;
+
+    if (kibanaSavedObjectMeta && kibanaSavedObjectMeta.searchSourceJSON && !savedSearchId) {
+      let searchSource;
+      try {
+        searchSource = JSON.parse(kibanaSavedObjectMeta.searchSourceJSON);
+        if (!searchSource.index) throw new Error('Missing index pattern');
```

Comment:
Throwing here means this function will return a promise sometimes and other times throw an error. I suggest `async` functions 😄